src/GraphQL.Tests/Types/ArgumentInformationTests.cs

```diff
@@ -0,0 +1,86 @@
+            {
+                Expression = null
+            };
+            info.Expression.ShouldBeNull();
+        }
+
+        [Fact]
+        public void Expression_Parses_Int()
+        {
+            var info = new ArgumentInformation(_testParameterInfo, typeof(object), new FieldType(), new TypeInformation(_testParameterInfo))
+            {
+                Expression = (Expression<Func<IResolveFieldContext, int>>)(context => 23)
+            };
+            info.Expression.Compile().DynamicInvoke(new object?[] { null }).ShouldBeOfType<int>().ShouldBe(23);
+        }
+
+        [Fact]
+        public void Expression_Parses_Object()
+        {
+            var info = new ArgumentInformation(_testParameterInfo, typeof(object), new FieldType(), new TypeInformation(_testParameterInfo))
+            {
+                Expression = (Expression<Func<IResolveFieldContext, object>>)(context => 23)
+            };
+            info.Expression.Compile().DynamicInvoke(new object?[] { null }).ShouldBeOfType<int>().ShouldBe(23);
+        }
+
+        [Fact]
+        public void Expression_Throws_For_Invalid_ObjectType()
+        {
+            var info = new ArgumentInformation(_testParameterInfo, typeof(object), new FieldType(), new TypeInformation(_testParameterInfo));
+            var error = Should.Throw<ArgumentException>(() => info.Expression = (Expression<Func<IResolveFieldContext, object>>)(context => "hello"));
+            error.Message.ShouldBe("Value must be a lambda expression delegate of type Func<IResolveFieldContext, Int32>.");
+        }
+
+        [Fact]
+        public void Expression_Throws_For_Invalid_Type()
+        {
+            var info = new ArgumentInformation(_testParameterInfo, typeof(object), new FieldType(), new TypeInformation(_testParameterInfo));
+            var error = Should.Throw<ArgumentException>(() => info.Expression = (Expression<Func<IResolveFieldContext, string>>)(context => "hello"));
+            error.Message.ShouldBe("Value must be a lambda expression delegate of type Func<IResolveFieldContext, Int32>.");
+        }
+
+        [Fact]
+        public void SetExpression_Parses_Int()
+        {
+            var info = new ArgumentInformation(_testParameterInfo, typeof(object), new FieldType(), new TypeInformation(_testParameterInfo));
+            info.SetExpression(context => 23);
+            info.Expression!.Compile().DynamicInvoke(new object?[] { null }).ShouldBeOfType<int>().ShouldBe(23);
+        }
+
+        [Fact]
+        public void SetExpression_Throws_For_Invalid_Type()
+        {
+            var info = new ArgumentInformation(_testParameterInfo, typeof(object), new FieldType(), new TypeInformation(_testParameterInfo));
+            var error = Should.Throw<ArgumentException>(() => info.SetExpression(context => "hello"));
+            error.Message.ShouldBe("Delegate must be of type Func<IResolveFieldContext, Int32>. (Parameter 'argumentDelegate')");
```

Comment:
```suggestion
            error.Message.ShouldStartWith("Delegate must be of type Func<IResolveFieldContext, Int32>");
```

Behaviour differs among TFMs.

Reply:
```suggestion
            error.Message.ShouldStartWith("Delegate must be of type Func<IResolveFieldContext, Int32>.");
```